### ISSUE TITLE
Error in dialogue_reader.gd: _evaluate_conditiona_expresion()

### ISSUE THREAD

**Issue by SpamFaux** [real-ezTheDev/GodotEzDialoguePlugin#10]
First, let me start in sharing our appreciation for your plug. Very cool and works really well. 

This error occurs when we attempt to perform conditional evaluations within a choice to create dynamic dialogue. While it seems as though this error does not impact the flow of the dialogue, we figured it would be worth letting you know. 

Example Evaluation:
`$if roll1 < 35 + Atlmod`

Error:
![image](https://github.com/real-ezTheDev/GodotEzDialoguePlugin/assets/26196070/79c55f38-5910-40d1-8354-1219cb94040c)

We had a further error where on occasion the dialogue reader would not / could not parse our state sent to the event. But at this point we believe this may be an issue with our `.godot` state getting corrupted with git. (Still looking into this.)

**Comment by real-ezTheDev** [comment]
Oh! Thanks for cutting this issue! 
I'll take a look as soon as I can this week!

**Comment by real-ezTheDev** [comment]
@SpamFaux 
Could you share a little more on the setup for this conditional.
for example, values of variable "roll1" and "Atlmod" and rest of the dialogue script in that dialogue node.

I'm trying to replicate the error with:
`$if roll1 < 35 + Atlmod`
but my tests are passing.

Also, are you running this with the most recent commit on the main?

*ps: regarding the state parsing... the plugin just assumes it's a Dictionary type and rely on Godot's handling of Dictionary variable so if there is a parsing issue, it does sound like it might be some sort of corruption issue.

**Comment by real-ezTheDev** [comment]
Ah i see, 
so sometimes you get that error messaging if the variables used in evaluation has no value.
However, functionally it still treats it as "false".

If this is the case for you, currently it'll just give you a harmless (non-flow-blocking) error messaging then evaluate it as false.
This maybe why you see the message but flow is not impacted.
I'll look into getting the error message not appear in such case. (or better warning message)

**Comment by SpamFaux** [comment]
I think this may have been more of a combination of an issue with how we were sending data through the state dictionary and actually processing the conditional within the dialogue event. We went through and fixed some items on our end resulting in the error getting cleared up. 

I've run through our established dialogues a ton now that we cleaned them up and I'm no longer getting the error. I'm betting we had something misaligned between the state being passed to the dialogue handler and what was actually getting evaluated in the conditional. 

Thank you very much for looking into this! Again, your plugin rocks.
